atomate/vasp/firetasks/surface_tasks.py

```diff
@@ -0,0 +1,174 @@
+# coding: utf-8
+
+from __future__ import division, print_function, unicode_literals, absolute_import
+
+import os
+
+from pymatgen.core import Structure
+from pymatgen.core.surface import SlabGenerator, ReconstructionGenerator, generate_all_slabs
+from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
+
+from fireworks.core.firework import FiretaskBase, FWAction
+from fireworks.utilities.fw_utilities import explicit_serialize
+
+
+"""
+Surface workflow firetasks.
+"""
+
+__author__ = "Richard Tran"
+__email__ = 'rit001@eng.ucsd.edu'
+
+
+@explicit_serialize
+class FacetFWsGeneratorTask(FiretaskBase):
+    """
+    Task for generating FWs for oriented unit cell calculations or slab
+    calculations. If the initial structure is an oriented unit cell,
+    generate slab fws, if its a conventional unit cell, generate oriented
+    unit cell and reconstruction calculations.
+    """
+
+    required_params = ['structure_type', "k_product", "vasp_cmd", "scratch_dir"]
+    optional_params = ["slab_gen_params", "mpid", "mmi", "db_file", "miller_index", "cwd"]
+
+    def run_task(self, fw_spec):
+        """
+        Generates a list of oriented unit cell FWs or
+        slab cell FWs depending on the structure type.
+
+        Args:
+            structure_type (str): Type of structure. Options are:
+                conventional_unit_cell, oriented_unit_cell and slab_cell
+            scratch_dir (str): - if specified, uses this directory as the root
+                scratch dir. Supports env_chk.
+            k_product (int): Default to 50, kpoint number * length for a & b
+                directions, also for c direction in bulk calculations.
+            db_file (str): FULL path to file containing the database credentials.
+                Supports env_chk.
+            vasp_cmd (str): Command to run vasp.
+            slab_gen_params (dict): Parameters for SlabGenerator or generate_all_slabs
+            mpid (str): Materials Project ID of the conventional unit cell.
+            mmi (int): Max Miller index.
+            miller_index ([h, k, l]): Miller index of plane parallel to
+                surface (and oriented unit cell).
+            cwd (str): Location of directory to operate the
```

Comment:
I find the param name ``cwd`` to be confusing since this stands for *current* working directory. Maybe something like ``run_dir`` or something would indicate to the user that this where the workflow will be run? Something else is fine too, just something that better describes the parameter